content/applications/productivity/mail_plugins/gmail.rst

```diff
@@ -2,94 +2,160 @@
+:guilabel:`Integrations`, activate :guilabel:`Mail Plugin`, and then click :guilabel:`Save`.
 
-#. Verify you are logged in using the Google account you want to install the plugin on.
+.. image:: gmail/mail-plugin-setting.png
+   :align: center
+   :alt: The Mail Plugin feature in the Settings.
 
-#. Click on *Publish* then *Deploy from manifest...*
+Configure the Gmail inbox
+-------------------------
 
-   .. image:: gmail/deploy-from-manifest.png
-      :align: center
-      :alt: Deploying from manifest the Gmail Plugin from the Apps Script project
+In the Gmail inbox, a purple Odoo icon is now visible on the right side panel. Click on the Odoo
+icon to open up the Odoo plugin window. Then, click on any email in the inbox. Click
+:guilabel:`Authorize Access` in the plugin window to grant Odoo access to the Gmail inbox.
 
-   .. important::
-      Make sure you are using the legacy editor; otherwise the *Deploy from manifest* functionality
-      may not be available.
+.. image:: gmail/authorize-access.png
+   :align: center
+   :alt: The Authorize Access button in the right sidebar of the Odoo plugin panel.
+
+Next, click :guilabel:`Login`. Then, enter the URL of the Odoo database that the user wishes to
+connect to the Gmail inbox, and log in to the database.
+
+.. note::
+   Use the general URL for the database, not the URL of a specific page in the database. For
+   example, use `https://mycompany.odoo.com`, not
+   `https://mycompany.odoo.com/web#cids=1&action=menu`.
+
+Finally, click :guilabel:`Allow` to let Gmail access the Odoo database. The browser will then show a
+:guilabel:`Success!` message. After that, close the window. The Gmail inbox and Odoo database are
+now connected.
+
+Odoo On-Premise users
+=====================
+
+For databases hosted on servers other than Odoo Online (or Odoo.sh), follow the steps below to
+configure the Gmail Plugin.
+
+.. note::
+   As part of their security guidelines, Google Workspace Marketplace requires add-on creators to
+   provide a list of URLs that can be used in actions (and redirections) launched by the add-on.
+   This protects users by ensuring, for example, that no add-on redirects users toward a malicious
+   website. (Read more on the `Google Apps Script
+   <https://developers.google.com/apps-script/manifest/allowlist-url>`_.)
+   
+   Since Odoo can only list the `odoo.com` domain and not every on-premise customer’s unique server
+   domain, on-premise customers cannot install the Gmail Plugin from Google Workspace Marketplace.
+
+Install the Gmail Plugin
+------------------------
 
-      .. image:: gmail/legacy-editor.png
-         :align: center
-         :alt: Using the legacy editor in the Gmail Plugin Apps Script project
+First, access the `GitHub repository <https://github.com/odoo/mail-client-extensions>`_ for the
+Odoo Mail Plugins. Next, click on the green :guilabel:`Code` button. Then click
+:guilabel:`Download ZIP` to download the Mail Plugin files onto the computer.
```

Comment:
```suggestion
Odoo Mail Plugins. Next, click on the green :guilabel:`Code` button. Then, click :guilabel:`Download
ZIP` to download the Mail Plugin files onto the computer.
```

Reply:
It's over 100 characters if I cut the line off after "Download", and the `guilabel` doesn't work if I cut it off after `:guilabel:`, so I cut it off after "click".